python/ray/train/torch.py

```diff
@@ -349,6 +396,55 @@ def __iter__(self):
             yield self._move_to_device(item)
 
 
+class _WrappedOptimizer(Optimizer):
+    def __init__(self, optimizer: Optimizer, scaler: Optional[GradScaler] = None):
+        self.optimizer = optimizer
+        self.scaler = scaler
+
+    @property
+    def state(self):
+        return self.optimizer.state
+
+    @state.setter
+    def state(self, state):
+        self.optimizer.state = state
+
+    @property
+    def param_groups(self):
+        return self.optimizer.param_groups
+
+    @param_groups.setter
+    def param_groups(self, param_groups):
+        self.optimizer.param_groups = param_groups
+
+    @property
+    def defaults(self):
+        return self.optimizer.defaults
+
+    @defaults.setter
+    def defaults(self, defaults):
+        self.optimizer.defaults = defaults
+
+    def add_param_group(self, param_group):
+        self.optimizer.add_param_group(param_group)
+
+    def load_state_dict(self, state_dict):
+        self.optimizer.load_state_dict(state_dict)
+
+    def state_dict(self):
+        return self.optimizer.state_dict()
+
+    def zero_grad(self):
```

Comment:
What are you guys thoughts on for both WrappedOptimizer and WrappedDataLoader, overriding `__getattr__` and just feeding everything except the overridden methods to the contained optimizer/dataloader?

Out of scope for this PR, but would like to hear your thoughts.

Reply:
Hmm...not sure. Benefit is fewer lines of code and potentially cleaner implementation. Downside is that the implementation is less explicit and potentially harder to read. 

My initial impression is that it'd be better to be explicit.